source/retryable-reads/tests/unified/handshakeError.yml

```diff
@@ -126,4 +127,681 @@ tests:
+        events:
+          - { connectionCheckOutStartedEvent: {} }
+          - { connectionCheckOutStartedEvent: {} }
+          - { connectionCheckOutStartedEvent: {} }
+          - { connectionCheckOutStartedEvent: {} }
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                ping: 1
+              databaseName: *databaseName
+          - commandStartedEvent:
+              command:
+                aggregate: *collectionName
+                pipeline:
+                  - $match: {}
+                  - $group: { _id: 1, n: { $sum: 1 } }
+              commandName: aggregate
+              databaseName: *databaseName
+
+  - description: "countDocuments succeeds after retryable handshake network error (ShutdownInProgress)"
+    operations:
+      - name: failPoint # fail the next connection establishment
+        object: testRunner
+        arguments:
+          client: *client0
+          failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 2 }
+            data:
+              failCommands: [saslContinue, ping]
+              errorCode: 91 # ShutdownInProgress
+
+      - name: runCommand
+        object: *database0
+        arguments:
+          commandName: ping
+          command: { ping: 1 }
+        expectError:
+          isError: true
+
+      - name: countDocuments
+        object: *collection0
+        arguments:
+          filter: {}
+        expectResult: 3
+
+    expectEvents:
+      - client: *client0
+        eventType: cmap
+        events:
+          - { connectionCheckOutStartedEvent: {} }
+          - { connectionCheckOutStartedEvent: {} }
+          - { connectionCheckOutStartedEvent: {} }
+          - { connectionCheckOutStartedEvent: {} }
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                ping: 1
+              databaseName: *databaseName
+          - commandStartedEvent:
+              command:
+                aggregate: *collectionName
+                pipeline:
+                  - $match: {}
+                  - $group: { _id: 1, n: { $sum: 1 } }
+              commandName: aggregate
+              databaseName: *databaseName
+
+  - description: "listDatabases succeeds after retryable handshake network error"
+    operations:
+      - name: failPoint # fail the next connection establishment
+        object: testRunner
+        arguments:
+          client: *client0
+          failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 2 }
+            data:
+              failCommands: [saslContinue, ping]
+              closeConnection: true
+
+      - name: runCommand
+        object: *database0
+        arguments:
+          commandName: ping
+          command: { ping: 1 }
+        expectError:
+          isError: true
+
+      - name: listDatabases
+        object: *client0
+        ignoreResultAndError: true # Assume success.
```

Comment:
```suggestion
```

IIUC the absence of `ignoreResultAndError` and `expectError` will [assert no error occurred](https://github.com/mongodb/specifications/blob/c796a746327af27f9070bd1791ced6bce9176453/source/unified-test-format/unified-test-format.rst#executing-an-operation).

> If operation.expectError is specified, the test runner MUST assert that the operation yielded an error; otherwise, the test runner MUST assert that the operation did not yield an error.

Suggested change applies to other operations with the `Assume success` comment.